src/main/java/org/apache/aurora/scheduler/storage/durability/ThriftBackfill.java

```diff
@@ -143,8 +148,28 @@ public IJobUpdate backFillJobUpdate(JobUpdate update) {
       backfillTask(instructions.getDesiredState().getTask());
     }
 
+    backfillUpdateStrategy(instructions.getSettings());
+
     instructions.getInitialState().forEach(e -> backfillTask(e.getTask()));
 
     return IJobUpdate.build(update);
   }
+
+  private void backfillUpdateStrategy(JobUpdateSettings settings) {
```

Comment:
I think we should also perform the reverse operation here if possible: If only the new settings are set, also write the deprecated ones.

This does not add much complexity but makes scheduler upgrade/downgrade paths easier to reason about.

Reply:
Will do this. In order to do this, I'll have a lossy conversion from the variable batch update strategy to a static batch update strategy which uses index 0 of the list contained by the variable batch update strategy.

e.g.: VarBatchUpdate = [1,2,3,4] will become BatchUpdate, group Size = 1 if the scheduler is downgraded to 0.19.0.

Reply:
It is more complicated than that. As soon as we persist one of these structs, we will be unable to roll back to any previous version of the software that does not have the thrift changes. To enable safe roll-back/roll-forward, we will need a commit/release with just the Thrift changes.

Reply:
Thanks for the input @DavidMcLaughlin, I'll create a PR with just the Thrift changes in that case.

Reply:
Tested this out by starting a vagrant box. Changed the code inside of [SchedulerThriftInterface.java](https://github.com/rdelval/aurora/blob/staggeredUpdates/src/main/java/org/apache/aurora/scheduler/thrift/SchedulerThriftInterface.java#L798) to set these fields upon receiving the request:

```    // Keep old job schema in case we want to revert to a lower version of Aurora that doesn't
    // support variable update group sizes
    int totalInstancesFromGroups;
    if (settings.getUpdateStrategy().isSetQueueStrategy()) {
      totalInstancesFromGroups = settings.getUpdateStrategy().getQueueStrategy().getGroupSize();
      settings.setWaitForBatchCompletion(false);
      settings.setUpdateGroupSize(settings.getUpdateStrategy().getQueueStrategy().getGroupSize());
    } else if (settings.getUpdateStrategy().isSetBatchStrategy()) {
      totalInstancesFromGroups = settings.getUpdateStrategy().getBatchStrategy().getGroupSize();
      settings.setWaitForBatchCompletion(true);
      settings.setUpdateGroupSize(settings.getUpdateStrategy().getBatchStrategy().getGroupSize());
    } else if (settings.getUpdateStrategy().isSetVarBatchStrategy()) {
      VariableBatchJobUpdateStrategy strategy = settings.getUpdateStrategy().getVarBatchStrategy();
      settings.setWaitForBatchCompletion(true);
      // Setting this value to allow graceful rollback to version 0.21.0
      // TODO(rdelvalle): Remove compatibility settings once version 0.22.0 ships
      settings.setUpdateGroupSize(1); // Necessary because we haven't validated fields yet.
```

I created a Variable batch update, then shut down the scheduler, and checked out tag rel/0.21.0 and ran aurorabuild all on the vagrant box. 

The scheduler came up fine. I forced a snapshot and a backup, restarted the scheduler with out issue. I upgraded the vagrant box back to the staggered updates patch and there was info loss (update no longer had staggered updates fields set so it stayed as a batch update), which in this case is an acceptable outcome IMO. 

Will update the patch with all the changes.